Cpp_test/lib/file-handler.cpp

```diff
@@ -0,0 +1,60 @@
+#include "file-handler.h"
+#include <ctime>
+#include <iostream>
+
+FileHandler::~FileHandler()
+{
+	if (fs.is_open())
+		fs.close();
+}
+
+unsigned long FileHandler::get_file_size()
+{
+	if (this->mode == FileMode::WRITE)
+	{
+		this->close_file();
+	}
+	return boost::filesystem::file_size(this->file_name);
+
+} 
+
+void FileHandler::open_file()
+{
+	if (this->mode == FileMode::READ)
```

Comment:
Another advice: you can have a specific typo rule to see variables within a class, like using _ at the end of the name. For instance here you can set the name of the variable `move_`, and with this you can call it whithout `this->` and still be easily readable.

Reply:
Thanks for your advice, I will do it in the next project.